extensions/azure/cosmos/assetindex-cosmos/src/test/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetIndexIntegrationTest.java

```diff
@@ -357,6 +360,21 @@ void findAll_withSorting() {
 
     }
 
+    @Test
+    void deleteById_whenPresent_deletes() {
+        Asset asset = createAsset(UUID.randomUUID().toString(), "test", "foobar");
+        container.createItem(new AssetDocument(asset, TEST_PARTITION_KEY, dataAddress));
+
+        Asset deletedAsset = assetIndex.deleteById(asset.getId());
+        assertThat(deletedAsset.getProperties()).isEqualTo(asset.getProperties());
+        assertThat(assetIndex.findById(asset.getId())).isNull();
+    }
+
+    @Test
+    void deleteById_whenMissing_returnsNull() {
+        assertThat(assetIndex.deleteById(UUID.randomUUID().toString())).isNull();
```

Comment:
nit: personally I think something like `"not-exists"` instead of the uuid makes it more clear what's going on

Reply:
Makes sense to me. I updated the various tests deleting a non existing item/asset to use the "not-exists" id.